src/main/java/me/jellysquid/mods/sodium/client/render/chunk/RenderSectionManager.java

```diff
@@ -82,10 +85,14 @@ public class RenderSectionManager {
     private @Nullable BlockPos lastCameraPosition;
 
     public RenderSectionManager(ClientWorld world, int renderDistance, CommandList commandList) {
-        this.chunkRenderer = new DefaultChunkRenderer(RenderDevice.INSTANCE, ChunkMeshFormats.COMPACT);
+        ChunkVertexType vertexType = SodiumClientMod.options().performance.useCompactVertexFormat ? ChunkMeshFormats.COMPACT : ChunkMeshFormats.VANILLA_LIKE;
```

Comment:
there's two places where the option is read and a vertex format selected. Could this maybe be put into some mechanism that supplies a `ChunkMeshFormat` based on the option and then we just read its data here and in `RenderRegion`? Also, there's at least three places I think where there's a case decision on each of the vertex formats. Maybe that could be reduced? I don't know how feasible that is and given that there probably won't be a bunch more vertex formats, it's unlikely to be very important, but it just feels kinda fragile.